extensions/artifacts/org.wso2.developerstudio.codenvy.ext.appfactory/src/main/java/org/wso2/developerstudio/codenvy/ext/appfactory/server/authentication/Authenticator.java

```diff
@@ -0,0 +1,64 @@
+/*
+* Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.wso2.developerstudio.codenvy.ext.appfactory.server.authentication;
+
+import org.wso2.developerstudio.codenvy.ext.appfactory.server.client.HTTPAppFactoryClient;
+import org.wso2.developerstudio.codenvy.ext.appfactory.server.exception.AppFactoryServerException;
+import org.wso2.developerstudio.codenvy.ext.appfactory.shared.dto.AppFactoryLoginInfo;
+import org.wso2.developerstudio.codenvy.ext.appfactory.shared.dto.AppFactoryLoginResponse;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class handles all the authentication related stuffs from WSO2 Developer Studio to WSO2 AppFactory/ AppCloud
+ */
+public class Authenticator {
```

Comment:
Why is this "singleton"? Any specific requirement that forced you to make it singleton?